legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabReviewApi.java

```diff
@@ -134,12 +132,24 @@ else if (revisionIds instanceof Set)
                     String mergeRequestStateString = mergeRequestState.toString();
                     mergeRequestStream = mergeRequestStream.filter(mr -> mergeRequestStateString.equalsIgnoreCase(mr.getState()));
                 }
+                if (sourceBranches.size() >= 1)
+                {
+                    mergeRequestStream = mergeRequestStream.filter(mr -> sourceBranches.contains(mr.getTargetBranch()));
+                }
             }
             else
             {
                 // if no revision ID is specified we will use the default merge request API from Gitlab to take advantage of the filter
                 MergeRequestFilter mergeRequestFilter = withMergeRequestFilters(new MergeRequestFilter(), state, since, until).withProjectId(gitLabProjectId.getGitLabId());
+                if (sourceBranches.size() == 1)
+                {
+                    mergeRequestFilter.withTargetBranch(sourceBranches.stream().findFirst().get());
+                }
                 mergeRequestStream = PagerTools.stream(withRetries(() -> getGitLabApi().getMergeRequestApi().getMergeRequests(mergeRequestFilter, ITEMS_PER_PAGE)));
+                if (sourceBranches.size() > 1)
```

Comment:
I would recommend adding this filter even in the case where there's just 1 source branch, just in case GitLab doesn't fully respect the MergeFilter. And given that you have the same filter logic in both branches of the main if, I'd unify the two cases.
```suggestion
                if (sourceBranches.notEmpty())
```